src/models/block.rs

```diff
@@ -61,34 +61,35 @@ impl From<EthBlockNumberOrTag> for StarknetBlockId {
     }
 }
 
+pub fn rpc_to_primitive_header(header: reth_rpc_types::Header) -> Result<reth_primitives::Header, KakarotError> {
+    return Ok(reth_primitives::Header {
+        base_fee_per_gas: header
+            .base_fee_per_gas
+            .map(|base_fee_per_gas| base_fee_per_gas.try_into().map_err(|_| ConversionError))
+            .transpose()?,
+        beneficiary: header.miner,
+        blob_gas_used: header.blob_gas_used.map(|blob_gas_used| blob_gas_used.to::<u64>()),
+        difficulty: header.difficulty,
+        excess_blob_gas: header.excess_blob_gas.map(|excess_blob_gas| excess_blob_gas.to::<u64>()),
+        extra_data: header.extra_data,
+        gas_limit: header.gas_limit.try_into().map_err(|_| ConversionError)?,
+        gas_used: header.gas_used.try_into().map_err(|_| ConversionError)?,
+        logs_bloom: header.logs_bloom,
+        mix_hash: header.mix_hash.unwrap_or_default(),
```

Comment:
Here are we sure that we should consider a default value if `None` because I believe for this specific example that it is an option because it is not always set on non-Ethereum networks. For me it's the same for the other items, I wonder if it wouldn't be better to also consider a conversion error if there is `None`?

Reply:
I think mix hash and nonce were just used for PoW no? so just setting a default value of 0 should be fine imo

Reply:
Yeah let us do like that and see if it induces some problems in the future. Maybe there will be a conflicting nonce between the real 0 nonce and the ones coming from None but should be fine as everything can be identified by other fields agreed?

Reply:
but I'm just not sure what we should put there otherwise :/ since nonce and mix hash are values that come from PoW ethereum, there isn't really any other value that we can set other than 0 right? or maybe default mix hash should be `keccak(0)` or something?

Reply:
I think this is fine at the moment, with the default implementation of `FixedBytes` this should output 0 for the mix hash

https://github.com/alloy-rs/core/blob/525a2339dfc963518a1350cf52e8928c50a49d66/crates/primitives/src/bits/fixed.rs#L37-L42

Maybe let us see and open a follow up PR if needed.